src/method.c

```diff
@@ -630,6 +630,7 @@ JL_DLLEXPORT jl_method_instance_t *jl_new_method_instance_uninit(void)
     jl_atomic_store_relaxed(&mi->cache, NULL);
     mi->cache_with_orig = 0;
     jl_atomic_store_relaxed(&mi->precompiled, 0);
+    mi->dispatched = 0;
```

Comment:
I think you want to reset this in staticdata too. Also should probably be made marked atomic

Reply:
As suggested by @timholy, I renamed `mi->precompiled` to `mi->flags`, made bit 0 mean `precompiled` and bit 1 mean `dispatched`.

So now `flags` are reset in staticdata, and are atomic.